crates/telnet-host/tests/common/mod.rs

```diff
@@ -0,0 +1,165 @@
+use assert_cmd::prelude::*;
+use std::{
+    io::{BufRead, BufReader, Write},
+    net::TcpStream,
+    path::{Path, PathBuf},
+    process::{Child, Command, Stdio},
+    sync::OnceLock,
+    thread::{self},
+    time::{Duration, Instant},
+};
+
+struct ManagedChild {
+    child: Child,
+}
+impl ManagedChild {
+    fn new(name: &'static str, mut child: Child) -> Self {
+        // Rust tests capture output, and hide it if the test passes unless `--nocapture` is passed to `cargo test`.
+        // This does *not* automatically apply to subprocesses, so: start threads to send subprocess output through
+        // `print!` / `eprintln!` to get the same behavior.
+        let stdout = child.stdout.take().expect("Failed to get stdout");
+        let stderr = child.stderr.take().expect("Failed to get stderr");
+        thread::spawn(|| {
+            let name = name.to_string();
+            let reader = BufReader::new(stdout);
+            for line in reader.lines() {
+                println!("[{name}]: {}", line.expect("Failed to read line"));
+            }
+        });
+        thread::spawn(|| {
+            let name = name.to_string();
+            let reader = BufReader::new(stderr);
+            for line in reader.lines() {
+                eprintln!("[{name}]: {}", line.expect("Failed to read line"));
+            }
+        });
+        Self { child }
+    }
+}
+impl Drop for ManagedChild {
+    fn drop(&mut self) {
+        self.child.kill().expect("Failed to kill child process");
+    }
+}
+
+fn start_daemon(db: &Path) -> ManagedChild {
+    let mut minimal_db = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    minimal_db.push("tests/Test.db");
+
+    ManagedChild::new(
+        "daemon",
+        Command::cargo_bin("moor-daemon")
+            .expect("Failed to find moor-daemon binary")
+            .arg("--textdump")
+            .arg(minimal_db)
+            .arg("--generate-keypair")
+            .arg("--max-buffer-pool-bytes")
+            .arg((1 << 24).to_string())
```

Comment:
Maybe toss this value into a constant with a comment

Reply:
Done!